pkg/resolver/search.go

```diff
@@ -294,25 +312,32 @@ func WhereClauseFilter(ctx context.Context, input *model.SearchInput,
 			if len(filter.Values) > 0 {
 				values := pointerToStringArray(filter.Values)
 
-				if len(propTypeMap) > 0 {
-					if dataTypeInMap, ok := propTypeMap[filter.Property]; !ok { // check if value exists
-						klog.Warningf("Property type for [%s] doesn't exist in cache. Refreshing property type cache",
-							filter.Property)
-						propTypeMapNew, err := getPropertyType(ctx, true) // Refresh the property type cache.
-						if err != nil {
-							klog.Warningf("Error creating property type map with err: [%s] ", err)
-						}
-						propTypeMap = propTypeMapNew
+				delete(propTypeMap, "kind")
+				dataType, dataTypeInMap := propTypeMap[filter.Property]
+				if len(propTypeMap) == 0 || !dataTypeInMap {
+					klog.Warningf("Property type for [%s] doesn't exist in cache. Refreshing property type cache",
+						filter.Property)
+					propTypeMapNew, err := getPropertyType(ctx, true) // Refresh the property type cache.
+					if err != nil {
+						klog.Errorf("Error creating property type map with err: [%s] ", err)
 						break
+					}
+					delete(propTypeMap, "kind")
```

Comment:
Why do we need to delete "kind"? It's also about 10 lines above.

Reply:
Oops, that was my debugging/recreating problem/testing code.